website/docs/interactors/4-matchers-assertions.md

```diff
@@ -0,0 +1,267 @@
+
+## Assertions
+
+There are four assertion methods available on all interactors:
+
+- `has()`
+- `is()`
+- `exists()`
+- `absent()`
+
+The assertion method you choose depends on what you are trying to test. Let's go through an example of each of these.
+
+:::note
+These different assertion methods are equivalents of Jest's `expect` and Cypress' `should`. When refactoring your test with Interactors, you would replace those constructs with the interactors' assertion methods.
+:::
+
+### Using `has()` to assert for one result
+
+If you expect that an interactor will only match one element, use `has()` to make your assertion.
+
+```js
+TextField({ id: 'username-id' }).has({ placeholder: 'USERNAME' });
+```
+
+### Using `is()` for readability
+
+The `is()` method is an alias of `has()`. It works the same way as `has()` The only difference is in the semantics so that your tests can read better.
+
+For instance, if we wanted to test if a text field is visible, `has()` would work perfectly fine; but writing the test using `is()` would read more like a sentence:
+
+```js
+TextField({ id: 'username-id' }).is({ visible: true });
+```
+
+As a rule of thumb, use `is()` if your assertion is an adjective, and use `has()` if your assertion is a noun. For example, if we wanted to assert for `visibility`, then it would sound more natural to use `has()` instead:
+
+```js
+TextField({ id: 'username-id' }).has({ visibility: true });
+```
+
+### Using `exists()` for multiple results
+
+Use `exists` when you expect that the interactor will find multiple results. This assertion will still pass even if there are multiple `TextField`s found.
+
+```js
+TextField().exists({ placeholder: 'USERNAME' });
+```
+
+### Using `absent()` for non-existant results
+
+If you want to assert that something is not visible on the page, use `absent()`. For example, when someone is logged in, maybe the sign in button should be hidden:
+
+```js
+Button('Sign In').absent();
+```
+
+## Writing flexible assertions using Matchers
+
+Matchers are methods that let you add logic or flexibility when you are searching for elements. They can be used within locators, filters, and assertions. In this section, you will learn how to use matchers to write better tests.
+
+Matchers include the following methods:
+
+- Text flexibility like `including()` and `matching()`
+- Conditionals like `and()`, `or()`, and `not()`
+- Iterables like `some()` and `every()`
+
+### When to use Matchers
+
+If your tests are written against a simulated database, it might not be important what the randomized users' names are. Maybe you just want to assert that a login success message includes the word "welcome", such as "Welcome, Taylor!" Or perhaps it does not matter what the name is, but you still want to assert that a name is displayed. These are situations where matchers would come in handy.
+
+### including, matching
```

Comment:
Would it be possible to have a parallel structure in this section for subtitles as in the previous ```Using `has()` to assert for one result``` etc?

Reply:
Could you elaborate?

Reply:
In the previous section, the functions were introduced with their function. For instance, the title is ```Using `absent()` for non-existant results``` instead of just `absent`

Reply:
This is how it is now:
![Screen Shot 2021-03-30 at 9 28 30 PM](https://user-images.githubusercontent.com/29791650/113077349-fa971e00-919e-11eb-85eb-2ef709271dfb.png)

And here's how it would look if we matched the pattern:
![Screen Shot 2021-03-30 at 9 28 43 PM](https://user-images.githubusercontent.com/29791650/113077369-0387ef80-919f-11eb-8f0c-01626274d969.png)

Maybe there's some better text to put in there instead of `String matchers:` and et cetera. But unlike assertions, which we're all already very familiar with, I think matching the pattern for the matcher section might not be necessary in my opinion. Because `String matchers: ..`, `Conditional matchers: ..` and `Iterable matchers: ..` don't really tell the users much until they actually read the content.

I mean at the same time I think the descriptive headings for the assertions aren't necessary either. It's crowding the sidebar and making it look more complicated.

Reply:
There are a few considerations here.

For longevity, it's good to not list function names in headings, so that as more methods get added to the library, or others get deprecated, your section headings don't change. This is important for things like keeping blog post links stable. We can see an example of generic naming in the Jest matchers docs: https://jestjs.io/docs/using-matchers

On the other hand, including the function names gives you a sense of what matchers are, and lets someone quickly find the docs when they are using it as a reference material. It's a nice escape hatch for when there is not a neat, descriptive heading to use.

As a general rule in Ember docs, I go with option #1. In shorter guides/articles, keeping links stable matters less. These articles are fairly short, especially if matchers get to be their own article in the future, and someone will not be lost of the heading disappears.

I think a good case could be made for either option. If I had to choose, in this case I'd personally go with using labels like "Iterable matchers" and not list the functions in the headings, just to allow for future growth and to keep the table of contents neat. Someone can command-F to search for the function names, and they are listed at the beginning of the section. That list could be links. I do think the proposed heading names like "iterable" and "conditional" have useful meaning.

Reply:
![Screen Shot 2021-04-01 at 10 12 51 AM](https://user-images.githubusercontent.com/29791650/113307200-e3f1e380-92d2-11eb-91df-c3adc7a0212d.png)

Reply:
@jorgelainfiesta If that looks good, can you resolve this conversation?